util/types/time.go

```diff
@@ -425,6 +412,32 @@ func (t *Time) check() error {
 	return nil
 }
 
+// Sub subtracts t1 from t, returns a duration value.
+// Note that sub should not be done on different time types.
+func (t *Time) Sub(t1 *Time) Duration {
+	var duration gotime.Duration
+	if t.Type == mysql.TypeTimestamp && t1.Type == mysql.TypeTimestamp {
+		a, _ := t.Time.GoTime()
+		b, _ := t1.Time.GoTime()
```

Comment:
Because it's safe to convert mysql.TypeTimestamp to go time, `t.Time.GoTime()` for mysql.TypeTimestamp won't return error because its month or day can't be 0.